### PR TITLE
Fix 240 - Update html files during publish if project metadata change…

### PR DESCRIPTION
**Pull request by lkisac**
Fixes Issue #240

### CODE REVIEW

File: api/classes/publisher.js [hunk 1]
```diff
@@ -310,6 +318,23 @@ class BasePublisher {
     return publishedFilesQueryBuilder
     .getAllModifiedFiles(this.publishedProject.id)
     .then(function(publishedFiles) {
+      if(!updateHtmlFilesWithNewRemixData) {
+        return publishedFiles;
+      }
+
+      var ignorePaths;
```
Comment (Pomax) [inline api/classes/publisher.js]: given that this value gets passed into `getAllFilesMatchPath` a few lines below, what should its default value be?

**Comment by flukeout** [comment]
Ping @gideonthomas or @cadecairos - either one of you want to review this?

File: api/classes/publisher.js [hunk 2]
```diff
@@ -236,6 +238,11 @@ class BasePublisher {
     return this.fetchPublishedProject()
     .then(function(publishedProject) {
```
Comment (gideonthomas) [inline api/classes/publisher.js]: change this to `.then(publishedProject => {`

File: api/classes/publisher.js [hunk 3]
```diff
@@ -236,6 +238,11 @@ class BasePublisher {
...
       if (publishedProject) {
+        // Check if project title/description is different from publishedProject title/description
+        // If it is, set flag to indicate that html files of publishedProject need to be updated with new remix metadata
+        if (project.title !== publishedProject.title || project.description !== publishedProject.description) {
+          updateHtmlFilesWithNewRemixData = true;
```
Comment (gideonthomas) [inline api/classes/publisher.js]: this should be `this.update...`

Comment (gideonthomas) [inline api/classes/publisher.js]: and change the flag to `remixDataChanged` instead.

File: api/classes/publisher.js [hunk 4]
```diff
@@ -310,6 +318,23 @@ class BasePublisher {
     return publishedFilesQueryBuilder
     .getAllModifiedFiles(this.publishedProject.id)
     .then(function(publishedFiles) {
```
Comment (gideonthomas) [inline api/classes/publisher.js]: same thing here:
`.then(publishedFiles => {`

and inside you can use `this.update...`

File: api/classes/publisher.js [hunk 5]
```diff
@@ -20,6 +20,8 @@ const publishedFilesQueryBuilder = require(`../modules/publishedFiles/model`).pr
 
 const ROOT_URL = `/`;
 
+var updateHtmlFilesWithNewRemixData = false;
```
Comment (gideonthomas) [inline api/classes/publisher.js]: this needs to be removed.

File: api/classes/publisher.js (hunk 4 above)
Comment (lkisac) [inline api/classes/publisher.js]: Thanks for the feedback @gideonthomas.  I've made the changes and tested, it's working well on my end.  Please let me know if there are any issues

**Comment by gideonthomas** [comment]
Hmm, looks like this fix didn't work.

I'm gonna debug to see why

File: api/classes/publisher.js [hunk 6]
```diff
@@ -285,6 +290,7 @@ class BasePublisher {
   }
 
   uploadModifiedFiles() {
+    const publishedProjectId = this.publishedProjectId;
```
Comment (gideonthomas) [inline api/classes/publisher.js]: this is incorrect! Look at how we access the publishedProject id below.

**Comment by lkisac** [comment]
Thanks for the review @gideonthomas   I've made the necessary changes

**Comment by gideonthomas** [comment]
@lkisac awesome thanks! Can you do one more small change? Can you make line 317 also use the variable you created for storing the published project id instead of accessing it directly?

**Comment by lkisac** [comment]
Thanks @gideonthomas  I've changed line 317.  Before I commit, I noticed some of the other methods below `deletePublishedProject` ([deletePublishedFiles](https://github.com/mozilla/publish.webmaker.org/blob/master/api/classes/publisher.js#L348-L359), [deleteOldFiles](https://github.com/mozilla/publish.webmaker.org/blob/master/api/classes/publisher.js#L361-L378)) are using `this.publishedProject.id`.  Should I set a `const publishedProjectId` for those methods as well?

**Comment by gideonthomas** [comment]
Let's keep this PR isolated to the changes for the issue it addresses so I don't think we should change those two methods.

**Comment by Pomax** [comment]
Yeah, always keep work isolated as much as possible: if you see a thing that needs to be fixed "as well", file a new issue that explicitly mentions those specific things, and then file a separate PR that addresses only those changes. This keeps things cleanly separated and easy to track.

**Comment by lkisac** [comment]
@Pomax  Absolutely.  It was something I noticed along side the last fix, while forgetting about the scope of the actual PR :/  I will file a new one for it, thanks!

**Comment by lkisac** [comment]
Awesome, thanks for the merge @gideonthomas!

**Pull request merged**